bleak/backends/winrt/client.py

```diff
@@ -0,0 +1,824 @@
+                                service,
+                                _communication_statues.get(
+                                    characteristics_result.status, ""
+                                ),
+                            )
+                        )
+                for characteristic in characteristics_result.characteristics:
+                    descriptors_result = await characteristic.get_descriptors_async(
+                        BluetoothCacheMode.CACHED
+                        if use_cached
+                        else BluetoothCacheMode.UNCACHED
+                    )
+                    self.services.add_characteristic(
+                        BleakGATTCharacteristicWinRT(characteristic)
+                    )
+                    if descriptors_result.status != GattCommunicationStatus.SUCCESS:
+                        if (
+                            characteristics_result.status
+                            == GattCommunicationStatus.PROTOCOL_ERROR
+                        ):
+                            raise BleakDotNetTaskError(
+                                "Could not get GATT descriptors for {0}: {1} (Error: 0x{2:02X}: {3})".format(
+                                    service,
+                                    _communication_statues.get(
+                                        descriptors_result.status, ""
+                                    ),
+                                    descriptors_result.protocol_error,
+                                    CONTROLLER_ERROR_CODES.get(
+                                        descriptors_result.protocol_error, "Unknown"
+                                    ),
+                                )
+                            )
+                        else:
+                            raise BleakDotNetTaskError(
+                                "Could not get GATT descriptors for {0}: {1}".format(
+                                    characteristic,
+                                    _communication_statues.get(
+                                        descriptors_result.status, ""
+                                    ),
+                                )
+                            )
+                    for descriptor in list(descriptors_result.descriptors):
+                        self.services.add_descriptor(
+                            BleakGATTDescriptorWinRT(
+                                descriptor,
+                                str(characteristic.uuid),
+                                characteristic.attribute_handle,
+                            )
+                        )
+
+            logger.info("Services resolved for %s", str(self))
+            self._services_resolved = True
+            return self.services
+
+    # I/O methods
+
+    async def read_gatt_char(
+        self,
+        char_specifier: Union[BleakGATTCharacteristic, int, str, uuid.UUID],
+        use_cached=False,
```

Comment:
type hint for `use_cached`? Or make it part of `**kwargs` like `get_services()`?

Reply:
Made it part of kwargs. It is still set to `False` by default and does not use not use the `self._use_cached` value. This is a bit problematic, but I *think* the users could be fooled if it is defaulted as `True` (as `self._use_cached` is by default)?

Reply:
Since we have lots of custom hardware users, False is definitely the best default.